tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -202,3 +218,404 @@ describe 'Annotator.Plugin.EnhancedAnchoring', ->
       assert.include hls, anchor12.highlight[anchor12.startPage]
       assert.include hls, anchor2.highlight[anchor2.startPage]
       assert.notInclude hls, anchor3.highlight[anchor3.startPage]
+
+  describe 'two-phased anchoring', ->
+
+    class DummyDocumentAccess
+
+      @applicable: -> true
+
+      isPageMapped: (index) -> index in @_rendered
+
+      constructor: ->
+        @_rendered = []
+
+    # Helper function to trigger a page rendering
+    renderPage = (doc, index) ->
+      return if doc.isPageMapped index
+
+      doc._rendered.push index
+
+      # Publish an event
+      event = document.createEvent "UIEvents"
+      event.initUIEvent "docPageMapped", false, false, window, 0
+      event.pageIndex = index
+      window.dispatchEvent event
+
+    # Helper function to trigger a page unrendering
+    unrenderPage = (doc, index) ->
+      return unless doc.isPageMapped index
```

Comment:
On 2015-02-10 15:27, Aron Carroll wrote:
>
> At least that way you get inconsistent behaviour instead of nothing
> happening. But a better way of dealing with bad input is just to throw
> an error if someone passes the wrong index.
>

Yes, that's indeed better.

I added exceptions instead.